### PR TITLE
Update to refactoring of ConvexShapeContact in hpp-constraints

### PR DESCRIPTION
**Pull request by florent-lamiraux**
Requires https://github.com/humanoid-path-planner/hpp-constraints/pull/117.

### CODE REVIEW

File: doc/Doxyfile.extra.in [hunk 1]
```diff
@@ -8,3 +8,5 @@ ALIASES   += Link{1}="\ref \1"
 ALIASES   += Link{2}="\ref \1 \"\2\""
 ALIASES   += LHPP{2}="\Link{hpp::\1::\2,\2}"
 ALIASES   += LPinocchio{1}="\LHPP{pinocchio,\1}"
+
+EXTRA_PACKAGES = {amsmath}
```
Comment (jmirabel) [inline doc/Doxyfile.extra.in:12]:
```suggestion
```

The cmake module add this and another package.

File: src/problem-solver.cc [hunk 2]
```diff
@@ -243,25 +244,15 @@ namespace hpp {
...
         if (!robot_->jointAndShapes.has (*it1))
           throw std::runtime_error ("First list of triangles not found.");
         l = robot_->jointAndShapes.get (*it1);
-        for (JointAndShapes_t::const_iterator it = l.begin ();
-            it != l.end(); ++it) {
-          constraints.first->addObject (ConvexShape (it->second, it->first));
+        for (JointAndShapes_t::const_iterator it(l.begin()); it!=l.end();++it)
+        {
+          objectSurfaces.push_back(*it);
         }
```
Comment (jmirabel) [inline src/problem-solver.cc:256]:
```suggestion
        objectSurfaces.insert(objectSurfaces.end(), l.begin(), l.end());
```

File: src/problem-solver.cc [hunk 3]
```diff
@@ -274,22 +265,38 @@ namespace hpp {
         else if (jointAndShapes.has (*it2))
           l = jointAndShapes.get (*it2);
         else throw std::runtime_error ("Second list of triangles not found.");
-        for (JointAndShapes_t::const_iterator it = l.begin ();
-            it != l.end(); ++it) {
-          constraints.first->addFloor (ConvexShape (it->second, it->first));
+        for (JointAndShapes_t::const_iterator it(l.begin()); it!=l.end();++it)
+        {
+          floorSurfaces.push_back(*it);
         }
```
Comment (jmirabel) [inline src/problem-solver.cc:271]:
```suggestion
        floorSurfaces.insert(floorSurfaces.end(), l.begin(), l.end());
```

File: src/problem-solver.cc [hunk 4]
```diff
@@ -326,15 +326,16 @@ namespace hpp {
...
           l = jointAndShapes.get (*it2);
         else throw std::runtime_error ("Second list of triangles not found.");
-
-        for (JointAndShapes_t::const_iterator it = l.begin ();
-            it != l.end(); ++it) {
-          cvxShape->addFloor (ConvexShape (it->second, it->first));
+        for (JointAndShapes_t::const_iterator it(l.begin()); it!=l.end();++it)
+        {
+          floorSurfaces.push_back(*it);
         }
```
Comment (jmirabel) [inline src/problem-solver.cc:332]:
```suggestion
        floorSurfaces.insert(floorSurfaces.end(), l.begin(), l.end());
```

File: src/problem-solver.cc [hunk 5]
```diff
@@ -298,22 +305,15 @@ namespace hpp {
...
           throw std::runtime_error ("First list of triangles not found.");
         l = robot_->jointAndShapes.get (*it1);
-
-        for (JointAndShapes_t::const_iterator it = l.begin ();
-            it != l.end(); ++it) {
-          cvxShape->addObject (ConvexShape (it->second, it->first));
+        for (JointAndShapes_t::const_iterator it(l.begin()); it!=l.end();++it)
+        {
+          objectSurfaces.push_back(*it);
         }
```
Comment (jmirabel) [inline src/problem-solver.cc:317]:
```suggestion
        objectSurfaces.insert(objectSurfaces.end(), l.begin(), l.end());
```

File: doc/Doxyfile.extra.in (hunk 1 above)
Comment (florent-lamiraux) [reply doc/Doxyfile.extra.in]: Done in 85b4f26.

File: src/problem-solver.cc (hunk 2 above)
Comment (florent-lamiraux) [reply src/problem-solver.cc]: Done in 85b4f26.

File: src/problem-solver.cc (hunk 3 above)
Comment (florent-lamiraux) [reply src/problem-solver.cc]: Done in 85b4f26.

File: src/problem-solver.cc (hunk 5 above)
Comment (florent-lamiraux) [reply src/problem-solver.cc]: Done in 85b4f26.

**Pull request merged**